src/SharpCompress/Readers/IReader.cs

```diff
@@ -8,6 +8,7 @@ public interface IReader : IDisposable
     {
         event EventHandler<ReaderExtractionEventArgs<IEntry>> EntryExtractionBegin;
```

Comment:
I would dump the Begin and End and only have Progress. They can check iterations or something to see if it's started.